changelog.d/pr-7220.md

```diff
@@ -0,0 +1,3 @@
+### 🏎 Performance
+
+- RF: get_modules_ - avoid O(N^2) if paths provided via use of get_limited_paths.  [PR #7220](https://github.com/datalad/datalad/pull/7220) (by [@yarikoptic](https://github.com/yarikoptic))
```

Comment:
doh -- we don't need changelog here since this is just an OPT on top of what we already have in master.
I better just adjust  http://github.com/datalad/datalad/blob/HEAD/changelog.d/pr-7189.md to reflect contribution/PRs and be done.  Will rewrite this commit